include/cuco/detail/bloom_filter.inl

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2021-2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuco/detail/hash_functions.cuh>
+#include <cuco/detail/utils.hpp>
+
+#include <cuda/std/atomic>
+
+#include <algorithm>
+#include <cstddef>
+#include <iterator>
+#include <memory>
+
+namespace cuco {
+
+template <typename Key, cuda::thread_scope Scope, typename Allocator, typename Slot>
+bloom_filter<Key, Scope, Allocator, Slot>::bloom_filter(std::size_t num_bits,
+                                                        std::size_t num_hashes,
+                                                        Allocator const& alloc)
+  : num_bits_{SDIV(std::max(std::size_t{1}, num_bits), detail::type_bits<slot_type>()) *
+              detail::type_bits<slot_type>()},
+    num_slots_{SDIV(std::max(std::size_t{1}, num_bits), detail::type_bits<slot_type>())},
+    num_hashes_{std::clamp(num_hashes, std::size_t{1}, detail::type_bits<slot_type>())},
+    slot_allocator_{alloc}
+{
+  slots_ = std::allocator_traits<slot_allocator_type>::allocate(slot_allocator_, num_slots_);
+
+  initialize();
```

Comment:
```suggestion
  initialize(stream);
```
Minor correction: this ctor should take a stream parameter and use it to initialize the map.